site/source/docs/porting/Debugging.rst

```diff
@@ -137,6 +137,58 @@ Debug printouts can even execute arbitrary JavaScript. For example::
   }
 
 
+Printing exception messages
+===========================
+
+Exceptions are thrown from WebAssembly using exception pointers, which means
+that try/catch/finally blocks in JavaScript will only receive a number, which
+represents a pointer into linear memory. In order to get the exception message,
+the user will need to create some WASM code which will extract the meaning from
+the exception. In the example code below we created a function that receives an
+``int`` which is pointer to an ``std::exception``, and by casting the pointer
+returns the ``what`` function call result.
+
+.. code-block:: cpp
+
+  #include <bind.h>
+
+  std::string getExceptionMessage(int exceptionPtr) {
```

Comment:
`int` could be `void*`

Reply:
When I try to use void* I get the following error when calling the function:

`Cannot call getExceptionMessage due to unbound types: Pv`

Reply:
Oh, interesting, I guess embind doesn't support that. But it should just work without embind.

As mentioned in another comment, I can't seem to get this to work with or without embind, so I must be doing something wrong - if you can provide a full testcase with an example that would be helpful. I can try to convert that to non-embind.

Reply:
I posted a link, below, for a test case - https://github.com/Shachlan/wasm-exception-example

Reply:
done